src/VisualStudio/Core/Def/Implementation/DesignerAttribute/VisualStudioDesignerAttributeService.cs

```diff
@@ -63,35 +64,7 @@ internal class VisualStudioDesignerAttributeService
 
         // We'll get notifications from the OOP server about new attribute arguments. Batch those
         // notifications up and deliver them to VS every second.
-        #region protected by lock
-
-        /// <summary>
-        /// Lock we will use to ensure the remainder of these fields can be accessed in a threadsafe
-        /// manner.  When OOP calls back into us, we'll place the data it produced into
-        /// <see cref="_updatedInfos"/>.  We'll then kick of a task to process this in the future if
-        /// we don't already have an existing task in flight for that.
-        /// </summary>
-        private readonly object _gate = new object();
-
-        /// <summary>
-        /// Data produced by OOP that we want to process in our next update task.
-        /// </summary>
-        private readonly List<DesignerInfo> _updatedInfos = new List<DesignerInfo>();
-
-        /// <summary>
-        /// Task kicked off to do the next batch of processing of <see cref="_updatedInfos"/>. These
-        /// tasks form a chain so that the next task only processes when the previous one completes.
-        /// </summary>
-        private Task _updateTask = Task.CompletedTask;
-
-        /// <summary>
-        /// Whether or not there is an existing task in flight that will process the current batch
-        /// of <see cref="_updatedInfos"/>.  If there is an existing in flight task, we don't need 
-        /// to kick off a new one if we receive more notifications before it runs.
-        /// </summary>
-        private bool _taskInFlight = false;
-
-        #endregion
+        private AsyncBatchingWorkQueue<DesignerInfo> _workQueue = null!;
```

Comment:
I'd prefer not cheating here and keeping this nullable. Then add `Contract.ThrowIfNull` in entry points that need this to be initialized.